mobile/examples/object_detection/ios/ORTObjectDetection/ModelHandler.swift

```diff
@@ -0,0 +1,448 @@
+            
+            // Translate the detected bounding box to CGRect.
+            rect.origin.y = CGFloat(detectionBoxes[4 * i])
+            rect.origin.x = CGFloat(detectionBoxes[4 * i + 1])
+            rect.size.height = CGFloat(detectionBoxes[4 * i + 2]) - rect.origin.y
+            rect.size.width = CGFloat(detectionBoxes[4 * i + 3]) - rect.origin.x
+            
+            let newRect = rect.applying(CGAffineTransform(scaleX: width, y: height))
+            
+            let colorToAssign = colorForClass(withIndex: detectionClassIndex + 1)
+            let inference = Inference(score: score,
+                                      className: detectionClass,
+                                      rect: newRect,
+                                      displayColor: colorToAssign)
+            resultsArray.append(inference)
+        }
+        
+        // Sort results in descending order of confidence.
+        resultsArray.sort { first, second -> Bool in
+            first.score > second.score
+        }
+        
+        return resultsArray
+    }
+    
+    // This method preprocesses the image by cropping pixel buffer to biggest square and scaling the cropped image to model dimensions.
+    private func preprocess(
+        ofSize size: CGSize,
+        _ buffer: CVPixelBuffer
+    ) -> CVPixelBuffer? {
+        let imageWidth = CVPixelBufferGetWidth(buffer)
+        let imageHeight = CVPixelBufferGetHeight(buffer)
+        let pixelBufferType = CVPixelBufferGetPixelFormatType(buffer)
+        
+        assert(pixelBufferType == kCVPixelFormatType_32BGRA ||
+            pixelBufferType == kCVPixelFormatType_32ARGB)
+        
+        let inputImageRowBytes = CVPixelBufferGetBytesPerRow(buffer)
+        let imageChannels = 4
+        
+        CVPixelBufferLockBaseAddress(buffer, CVPixelBufferLockFlags(rawValue: 0))
+        
+        // Find the biggest square in the pixel buffer and advance rows based on it.
+        guard let inputBaseAddress = CVPixelBufferGetBaseAddress(buffer) else {
+            return nil
+        }
+        
+        // Get vImage_buffer
+        var inputVImageBuffer = vImage_Buffer(
+            data: inputBaseAddress, height: UInt(imageHeight), width: UInt(imageWidth),
+            rowBytes: inputImageRowBytes
+        )
+        
+        let scaledRowBytes = Int(size.width) * imageChannels
+        guard let scaledImageBytes = malloc(Int(size.height) * scaledRowBytes) else {
+            return nil
+        }
+        var scaledVImageBuffer = vImage_Buffer(data: scaledImageBytes, height: UInt(size.height), width: UInt(size.width), rowBytes: scaledRowBytes)
+        
+        // Perform the scale operation on input image buffer and store it in scaled vImage buffer.
+        let scaleError = vImageScale_ARGB8888(&inputVImageBuffer, &scaledVImageBuffer, nil, vImage_Flags(0))
+        
+        CVPixelBufferUnlockBaseAddress(buffer, CVPixelBufferLockFlags(rawValue: 0))
```

Comment:
>CVPixelBufferUnlockBaseAddress [](http://example.com/codeflow?start=8&length=30)

should we call this from a defer block?
as a side note, I found we can introduce a new scope with `do { ... }` to run the defer block sooner if needed